src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -5369,8 +5475,9 @@ public void closeDiscardedServerPreparedStatements() {
      * Remove references to outstanding un-prepare requests. Should be run when connection is closed.
      */
     private final void cleanupPreparedStatementDiscardActions() {
+        closeDiscardedServerPreparedStatements();
```

Comment:
There is no need to call close here as all handles will be discarded by the server when the connection is closed.